configure

```diff
@@ -672,40 +685,29 @@ def configure_node(o):
     o['variables']['node_target_type'] = 'static_library'
 
 
-def configure_libz(o):
-  o['variables']['node_shared_zlib'] = b(options.shared_zlib)
-
-  if options.shared_zlib:
-    o['libraries'] += ['-l%s' % options.shared_zlib_libname]
-  if options.shared_zlib_libpath:
-    o['libraries'] += ['-L%s' % options.shared_zlib_libpath]
-  if options.shared_zlib_includes:
-    o['include_dirs'] += [options.shared_zlib_includes]
-
+def configure_library(lib, output):
+  shared_lib = 'shared_' + lib
+  output['variables']['node_' + shared_lib] = b(getattr(options, shared_lib))
 
-def configure_http_parser(o):
-    o['variables']['node_shared_http_parser'] = b(options.shared_http_parser)
+  if getattr(options, shared_lib):
+    default_cflags = getattr(options, shared_lib + '_includes')
+    default_lib = format_libraries(getattr(options, shared_lib + '_libname'))
+    default_libpath = getattr(options, shared_lib + '_libpath')
+    if default_libpath:
+      default_libpath = '-L' + default_libpath
+    (pkg_libs, pkg_cflags, pkg_libpath) = pkg_config(lib)
+    cflags = pkg_cflags.split('-I') if pkg_cflags else default_cflags
```

Comment:
The problem with splitting with space is that the first item won't strip its `-I`. I'd have to do some extra logic for that to work as intended. I think `-I` is rare enough. If we get a bug filed, I'll revisit.